src/components/AccordionMenu/index.tsx

```diff
@@ -0,0 +1,98 @@
+import * as React from 'react';
+import { Button } from 'react-bootstrap';
+import { connect } from 'react-redux';
+import makeClassName from 'classnames';
+
+import { ApplicationState } from '../../reducers';
+import { ConnectedReduxProps } from '../../configureStore';
+import { actions, isExpanded } from '../../reducers/accordionMenu';
+import { AnyReactNode } from '../../typeUtils';
+import styles from './styles.module.scss';
+
+export const makeItemContentId = (title: string) => {
+  const id = title.replace(/[^A-Za-z0-9]+/g, '');
+  if (!id) {
+    throw new Error(`An ID could not be generated from this title: "${title}"`);
+  }
+  return `accordionContent${id}`;
+};
+
+export type PublicItemProps = {
+  children: AnyReactNode;
+  title: string;
+};
+
+export type DefaultItemProps = {
+  expanded: boolean;
+};
+
+type ItemPropsFromState = {
+  expandedFromState: boolean;
+};
+
+type ItemProps = PublicItemProps &
+  DefaultItemProps &
+  ItemPropsFromState &
+  ConnectedReduxProps;
+
+export class AccordionItemBase extends React.Component<ItemProps> {
+  static defaultProps = {
+    expanded: false,
+  };
+
+  componentDidMount() {
+    const { dispatch, expanded, expandedFromState, title } = this.props;
+
+    if (expanded && !expandedFromState) {
+      dispatch(actions.toggleItem({ itemId: title }));
+    }
+  }
+
+  onItemClick = () => {
+    const { dispatch, title } = this.props;
+    dispatch(actions.toggleItem({ itemId: title }));
+  };
+
+  render() {
+    const { children, expandedFromState, title } = this.props;
+    const contentId = makeItemContentId(title);
+
+    return (
+      <React.Fragment>
+        <Button
+          aria-controls={contentId}
+          onClick={this.onItemClick}
+          className={makeClassName(styles.item, styles.itemButton)}
+        >
+          {title}
+        </Button>
+        <div
+          aria-expanded={expandedFromState ? 'true' : 'false'}
+          className={makeClassName(styles.item, styles.itemContent, {
```

Comment:
There don't seem to be any tests that assert that the `div` has a class of `styles.item`. Does that matter, or is that not something we'd usually test for?

Reply:
Same answer as above. It doesn't matter because this is configuration that won't change based on inputs. A test would only fail if the developer intentionally removed the configuration which would just annoy the developer at best.